### PR TITLE
Bm email set admin dash

### PR DESCRIPTION
**Pull request by brunomembrado**

### CODE REVIEW

File: src/app/api/generateData/orders/route.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  if (!user || !user.isAdmin) {
+    return Response.json(
+      { message: "unauthorized" },
+      {
+        status: 401,
+      }
+    );
+  }
+
+  await dbConnect();
```
Comment (andvirga) [inline src/app/api/generateData/orders/route.ts:23]: We need to change this to use a Singleton class, in that way we won't instantiate a new connection per each request. We can do this on a separate PR / ticket.

File: src/app/api/generateData/products/route.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Save products with incremental productNumber sequentially
+  for (const product of products) {
+    const newProduct = new ProductModel(product);
+    await newProduct.save();
+    await new Promise((res, rej) => {
+      setTimeout(() => {
+        res("");
+      }, 10);
+    });
```
Comment (andvirga) [inline src/app/api/generateData/products/route.ts]: Is this delay intentional?

Comment (andvirga) [reply src/app/api/generateData/products/route.ts]:
```suggestion
```

You can safely remove this code, you're already handling the update counter logic here:

https://github.com/solvd-interviews/bench-ecommerce-app-v3/blob/a2bfe913bbd6b30d25982ca6254979011526a75d/src/lib/models/ProductModel.ts#L54

File: src/app/api/generateData/users/route.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  // Save users with incremental userNumber sequentially
+  for (const user of users) {
+    const newUser = new UserModel(user);
+    await newUser.save();
+    await new Promise((res, rej) => {
+      setTimeout(() => {
+        res("");
+      }, 10);
+    });
```
Comment (andvirga) [inline src/app/api/generateData/users/route.ts]:
```suggestion
```

Same explanation as previous comment

**Review by andvirga — APPROVED** [review]
LGTM! Great job.

Only left some comments to remove unnecessary delays.

I'll create a new ticket to replace ` await dbConnect();` with a Singleton class, so we don't open a new DB Connection on each API request (anti-pattern)

File: src/app/api/generateData/products/route.ts (hunk 2 above)
Comment (brunomembrado) [reply src/app/api/generateData/products/route.ts]: Yes Andres the delay is intentional in order to have a createdAt ordered. I remember that they bugged at the table, and this solved the problem.
